doc/user/formats.rst

```diff
@@ -0,0 +1,129 @@
+actual model. Aliases can also point at other aliases, but there is an
+implementation-defined maximum redirection depth. Aliases may not change
+origin (scheme, host or port); this makes it practical to mirror a set of
+models in a filesystem or on another server without needing to rewrite the
+aliases.
+
+Model files will contain the following attributes on the root (the ``model_*``
+namespace is reserved for the general framework, and shouldn't be used for
+type-specific data):
+
+model_type
+    The model type (see :doc:`concepts`). See later sections for model types
+    defined by this specification. New model types may be defined in future.
+    To avoid collisions, users may define their own types with the prefix
+    ``x_``.
+
+model_format
+    Indicates how the model is encoded in the file, and is specific to each
+    model type. This value also serves as a major version number: if
+    backwards-incompatible changes are made, a new ``model_format`` must be used.
+    There is no minor version numbering; just add new datasets, attributes
+    etc.
+
+model_target (optional)
+    Human-readable string describing the target of the model. Note that
+    because models may be shared (e.g., multiple receptors may have the same
+    SEFD model), this might be more general than the target the user
+    requested.
+
+model_comment (optional)
+    Human-readable string which may describe the provenance of the model or
+    any other comments about it.
+
+model_created (optional)
+    Timestamp at which the model was created, in RFC 3339 format.
+
+model_version:
+    Simple integer version, incremented only when semantic content changes to
+    the model occur. Thus, ``model_format`` may be updated without changing
+    this version, provided that the new parameters describe the same model.
+
+model_author (optional):
+    Name of the person who created the model. It may optionally include an
+    email address in angle brackets (name-addr format in RFC 2822).
+
+Physical units are unscaled. In other words, frequencies are in units of Hz,
+flux density in units Jy and so on. Angles are in radians, or expressed as
+direction cosines when appropriate.
+
+RFI Mask
+--------
+This specifies frequencies which are affected by Radio Frequency
+Interference (RFI). It is assumed that the RFI environment is consistent
+across a whole telescope.
+
+Attributes
+^^^^^^^^^^
+model_type
+    ``rfi_mask``
+
+model_format
+    ``ranges``
+
+mask_auto_correlations
+    A boolean to indicate whether autocorrelations (both same-hand and
+    cross-hand) should be masked, in which case it is done for frequencies
+    covered by any of the ranges.
+
+Datasets
+^^^^^^^^
+ranges
+    A 1D dataset with the following columns
+
+    min_frequency, max_frequency (float)
+        Frequencies of an RFI-affected region of the spectrum. Channels are
+        masked if the channel centre frequency falls into any of the ranges
```

Comment:
It is therefore possible to include RFI in <50% of a channel that straddles the endpoint of a range. To avoid this, one has to extend the range by half a channel width during the checks, but maybe this is not worth the effort.

Reply:
That's a good point. Unfortunately fixing it will require an API change, because the current API doesn't have any information about channel width. I've filed https://skaafrica.atlassian.net/browse/SPR1-740 for it.